openedx/core/djangoapps/auth_exchange/views.py

```diff
@@ -171,7 +185,15 @@ def post(self, request):
         # The login method assumes the backend path had been previously stored in request.user.backend
         # in the 'authenticate' call.  However, not all authentication providers do so.
         # So we explicitly populate the request.user.backend field here.
+
         if not hasattr(request.user, 'backend'):
             request.user.backend = self._get_path_of_arbitrary_backend_for_user(request.user)
+
+        if not self._is_grant_password(request.auth):
+            raise AuthenticationFailed({
```

Comment:
You are missing a test for this - to make sure `AuthenticationFailed` exception gets raised when it should.

Reply:
I think 401 status code will cover this.

Reply:
I see.  The test method name confused me since it says `test_dot_unauthenticated`, which I thought meant that the request itself was unauthenticated.  Instead, it was authenticated but with client-credentials grant.